src/synopsys-action/tools-parameter.ts

```diff
@@ -82,61 +82,64 @@ export class SynopsysToolsParameter {
-    if (parseToBoolean(inputs.POLARIS_PRCOMMENT_ENABLED)) {
-      info('Polaris PR comment is enabled')
-      if (inputs.POLARIS_PARENT_BRANCH_NAME) {
-        polData.data.polaris.branch.parent.name = inputs.POLARIS_PARENT_BRANCH_NAME
-      }
-      const prCommentSeverities: string[] = []
-      const inputPrCommentSeverities = inputs.POLARIS_PRCOMMENT_SEVERITIES
-      if (inputPrCommentSeverities != null && inputPrCommentSeverities.length > 0) {
-        const severityValues = inputPrCommentSeverities.split(',')
-        for (const severity of severityValues) {
-          if (severity.trim()) {
-            prCommentSeverities.push(severity.trim())
+    if (isPullRequestEvent()) {
+      /** Set Polaris PR comment inputs in case of PR context */
+      if (parseToBoolean(inputs.POLARIS_PRCOMMENT_ENABLED)) {
+        info('Polaris PR comment is enabled')
+        if (inputs.POLARIS_PARENT_BRANCH_NAME) {
+          polData.data.polaris.branch.parent.name = inputs.POLARIS_PARENT_BRANCH_NAME
+        }
+        const prCommentSeverities: string[] = []
+        const inputPrCommentSeverities = inputs.POLARIS_PRCOMMENT_SEVERITIES
+        if (inputPrCommentSeverities != null && inputPrCommentSeverities.length > 0) {
+          const severityValues = inputPrCommentSeverities.split(',')
+          for (const severity of severityValues) {
+            if (severity.trim()) {
+              prCommentSeverities.push(severity.trim())
+            }
           }
         }
+        polData.data.polaris.prComment = {
+          enabled: true,
+          severities: prCommentSeverities
+        }
+        polData.data.github = this.getGithubRepoInfo()
       }
-      polData.data.polaris.prComment = {
-        enabled: true,
-        severities: prCommentSeverities
-      }
-      polData.data.github = this.getGithubRepoInfo()
     } else {
-      polData.data.polaris.prComment = {enabled: false}
-    }
-    if (!isPullRequestEvent() && parseToBoolean(inputs.POLARIS_REPORTS_SARIF_CREATE)) {
-      const sarifReportFilterSeverities: string[] = []
-      const sarifReportFilterAssessmentIssuesType: string[] = []
-
-      if (inputs.POLARIS_REPORTS_SARIF_SEVERITIES) {
-        const filterSeverities = inputs.POLARIS_REPORTS_SARIF_SEVERITIES.split(',')
-        for (const fixPrSeverity of filterSeverities) {
-          if (fixPrSeverity != null && fixPrSeverity !== '') {
-            sarifReportFilterSeverities.push(fixPrSeverity.trim())
+      /** Set Polaris SARIF inputs in case of non PR context */
+      if (parseToBoolean(inputs.POLARIS_REPORTS_SARIF_CREATE)) {
+        const sarifReportFilterSeverities: string[] = []
+        const sarifReportFilterAssessmentIssuesType: string[] = []
+
+        if (inputs.POLARIS_REPORTS_SARIF_SEVERITIES) {
+          const filterSeverities = inputs.POLARIS_REPORTS_SARIF_SEVERITIES.split(',')
+          for (const sarifSeverity of filterSeverities) {
+            if (sarifSeverity != null && sarifSeverity !== '') {
```

Comment:
We can put condition like 'if(sarifSeverity)". This way condition checks for null, empty or undefined variable.

Reply:
updated